dataikuapi/dss/apideployer.py

```diff
@@ -191,6 +200,18 @@ def add_apinode(self, url, api_key, graphite_prefix=None):
         }
         self.settings["apiNodes"].append(new_node)
 
+    def remove_apinode(self, node_url):
+        """
+        Removes a node from the list of nodes of this infra.
+        Only applicable to STATIC infrastructures
+
+        :param str node_url: URL of the node to remove
+        """
+        for node in self.settings["apiNodes"]:
+            if node["url"] == node_url:
+                self.settings["apiNodes"].remove(node)
+                break
```

Comment:
To discuss: maybe do not break here to remove all the nodes which url matches. There is no unique check on URL of apiNodes

Reply:
Oops, I was persuaded URL was unique 🙂